core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1150,7 +1150,7 @@ class DAGScheduler(
       case TaskResultLost =>
         // Do nothing here; the TaskScheduler handles these failures and resubmits the task.
 
-      case other =>
+      case ExecutorLostFailure(_) | TaskKilled | UnknownReason =>
```

Comment:
this change is so I could trigger an exception (by passing in `null`), seems like a good change in any case.